client/src/partials/reports/stock_integration/stock_integration.js

```diff
@@ -0,0 +1,150 @@
+
+    $scope.selected = null;
+
+    var dependencies = {};
+
+    dependencies.projects = {
+      required: true,
+      query : {
+        tables : {
+          'project' : {
+            columns : ['id', 'abbr', 'name']
+          }
+        }
+      }
+    };
+
+    dependencies.depots = {
+      required: true,
+      query : {
+        tables : {
+          'depot' : {
+            columns : ['uuid', 'text', 'reference', 'enterprise_id']
+          }
+        }
+      }
+    };
+
+    function day () {
+      session.dateFrom = new Date();
+      session.dateTo = new Date();
+    }
+
+    function week () {
+      session.dateFrom = new Date();
+      session.dateTo = new Date();
+      session.dateFrom.setDate(session.dateTo.getDate() - session.dateTo.getDay());
+    }
+
+    function month () {
+      session.dateFrom = new Date();
+      session.dateTo = new Date();
+      session.dateFrom.setDate(1);
+    }
+
+
+    dependencies.currencies = {
+      required : true,
+      query : {
+        tables : {
+          'currency' : {
+            columns : ['id', 'symbol']
+          }
+        }
+      }
+    };
+
+    $scope.options = [
+      {
+        label : 'CASH_PAYMENTS.DAY',
+        fn : day,
+      },
+      {
+        label : 'CASH_PAYMENTS.WEEK',
+        fn : week,
+      },
+      {
+        label : 'CASH_PAYMENTS.MONTH',
+        fn : month
+      }
+    ];
+
+    function search (selection) {
+      session.selected = selection.label;
+      selection.fn();
+    }
+
+    function reset (p) {
+      session.searching = true;
+      var req, url;
+
+      $scope.state = 'generate';
+      // toggle off active
+      session.active = !p;
+
+      req = {
+        dateFrom : session.dateFrom,
+        dateTo : session.dateTo,
+        depot : session.depot_entry
+      };
+
+      url = '/reports/integration_stock/?start=%start%&end=%end%&depot=%depot%'
+      .replace('%start%', req.dateFrom)
+      .replace('%end%', req.dateTo)
+      .replace('%depot%', req.depot);
+
+      connect.fetch(url)
+      .then(function (model) {
+        if (!model) { return; }
+        $scope.integration_records = model;
+      });
+
+    }
+
+    appstate.register('project', function (project) {
+      session.project = project.id;
+      validate.process(dependencies)
+      .then(function (models) {
+        $scope.depots = models.depots.data;
+        $scope.projects = models.projects;
+        $scope.currencies = models.currencies;
+        $scope.projectAbbr = session.project.abbr;
+        session.currency = $scope.currencies.data[0].id;
+        $scope.allProjectIds =
+          models.projects.data.reduce(function (a,b) { return a + ',' + b.id ; }, '')
+          .substr(1);
+        search($scope.options[0]);
+      });
+    });
+
+    appstate.register('enterprise', function (enterprise) {
+      $scope.enterprise = enterprise;
+    });
+
+    $scope.print = function print() {
+      window.print();
```

Comment:
To call native functions, it is best to use [Angular's `$window` service](https://docs.angularjs.org/api/ng/service/$window).